astropy/coordinates/matching.py

```diff
@@ -299,6 +315,14 @@ def search_around_sky(coords1, coords2, seplimit, storekdtree='_kdtree_sky'):
     if not seplimit.isscalar:
         raise ValueError('seplimit must be a scalar in search_around_sky')
 
+    if coords1.isscalar or coords2.isscalar:
+        raise ValueError('One of the inputs to search_around_sky is a scalar. '
+                         'search_around_sky is intended for use with array '
+                         'coordinates, not scalars.  Use the `separation` '
+                         'method on coordinate objects along with '
+                         '`numpy.argmin` to find the closest match to a scalar'
+                         'coordinate.')
```

Comment:
You're suggesting to use `separation` and `numpy.argmin` to find the closest match in the `search_around_sky` function, which is about finding all matches within a given separation, so it's not what the user wanted.

After thinking about this some more, I'm not so sure any more if it wouldn't be better to make the scalar case just work here or to advice users to call the SkyCoord equivalent of `numpy.atleast_1d` (which I still don't know if it exists or what it is) before calling `search_around_sky` instead of suggesting they re-implement `search_around_sky` e.g. for the case where `coords1` is scalar and `coords2` has 100 elements and the user wants the index array of those within `seplimit`.